npm/src/upload/index.ts

```diff
@@ -0,0 +1,76 @@
+import { TdrFile } from "@nationalarchives/file-information"
+import { ClientFileProcessing } from "../clientprocessing"
+
+interface HTMLInputTarget extends EventTarget {
+  files?: InputElement
+}
+
+interface InputElement {
+  files?: TdrFile[]
+}
+
+export class UploadFiles {
+  clientFileProcessing: ClientFileProcessing
+
+  constructor(clientFileProcessing: ClientFileProcessing) {
+    this.clientFileProcessing = clientFileProcessing
+  }
+
+  upload(): void {
+    const uploadForm: HTMLFormElement | null = document.querySelector(
+      "#file-upload-form"
+    )
+
+    if (uploadForm) {
+      uploadForm.addEventListener("submit", ev => {
+        ev.preventDefault()
+        const consignmentId: number = this.retrieveConsignmentId()
+
+        if (!consignmentId) {
+          throw Error("No consignment provided")
+        }
+
+        const target: HTMLInputTarget | null = ev.currentTarget
+
+        try {
+          const files: TdrFile[] = target!.files!.files!
+
+          this.generateFileDetails(consignmentId, files.length).then(r => {
+            this.uploadClientFileMetadata(r, files)
+          })
+        } catch (e) {
+          //For now console log errors
+          console.log("Upload failed: " + e.message)
+        }
+      })
+    }
+  }
+
+  retrieveConsignmentId(): number {
+    const pathName: string = window.location.pathname
+    const paths: string[] = pathName.split("/", 3)
+
+    return parseInt(paths[2]!, 10)
+  }
+
+  //Split to separate function to make testing easier
```

Comment:
You could actually just split the functions out into a different module. It makes sense to separate the HTML-wrangling code above from the lower-level code that calls the API, and then it's also easier to test.